pkg/cluster/fixetcd.go

```diff
@@ -0,0 +1,517 @@
+	err = kubecli.BatchV1().Jobs(b.Namespace).Delete(ctx, jobNameFixPeers, metav1.DeleteOptions{
+		PropagationPolicy: &propPolicy,
+	})
+	if err != nil {
+		return err
+	}
+
+	// return errors from deferred cleanup function
+	return err
+}
+
+func createPrivilegedServiceAccount(ctx context.Context, log *logrus.Entry, name, usersAccount string, kubecli kubernetes.Interface, securitycli securityv1client.SecurityContextConstraintsInterface) (func() error, error) {
+	log.Infof("Creating service account %s now", name)
+	serviceAcc, err := kubecli.CoreV1().ServiceAccounts(nameSpaceEtcds).Create(ctx, &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		AutomountServiceAccountToken: to.BoolPtr(true),
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Creating cluster role %s now", usersAccount)
+	cr, err := kubecli.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: usersAccount,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{"get", "create"},
+				Resources: []string{"pods", "pods/exec"},
+				APIGroups: []string{""},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Creating cluster role binding %s now", serviceAccountName)
+	crb, err := kubecli.RbacV1().ClusterRoleBindings().Create(ctx, &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: serviceAccountName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     kubeServiceAccount,
+			APIGroup: "",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccountName,
+				Namespace: nameSpaceEtcds,
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Creating Security Context Constraint %s now", name)
+	scc, err := securitycli.Create(ctx, &securityv1.SecurityContextConstraints{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: nameSpaceEtcds,
+		},
+		Groups:                   []string{},
+		Users:                    []string{usersAccount},
+		AllowPrivilegedContainer: true,
+		AllowPrivilegeEscalation: to.BoolPtr(true),
+		AllowedCapabilities:      []corev1.Capability{"*"},
+		RunAsUser: securityv1.RunAsUserStrategyOptions{
+			Type: securityv1.RunAsUserStrategyRunAsAny,
+		},
+		SELinuxContext: securityv1.SELinuxContextStrategyOptions{
+			Type: securityv1.SELinuxStrategyRunAsAny,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return func() error {
+		log.Infof("Deleting service account %s now", serviceAcc.Name)
+		err := kubecli.CoreV1().ServiceAccounts(nameSpaceEtcds).Delete(ctx, serviceAcc.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+
+		log.Infof("Deleting %s now", scc.Name)
```

Comment:
Did you intend for this log message to be `"Deleting security context constraint %s now"`?

Reply:
Yes, thank you.